src/parthenon_manager.cpp

```diff
@@ -148,6 +148,10 @@ void ParthenonManager::ParthenonInitPackagesAndMesh() {
 
     // Read package data from restart file
     RestartPackages(*pmesh, *restartReader);
+
+    // close hdf5 file to prevent HDF5 hangs and corrupted files
+    // if code dies after restart
+    this->restartReader = nullptr;
```

Comment:
I don't understand why this fixes a bug, but I'll take your word for it.  Why explicitly use `this->` though?

Reply:
I copied it from the destructor. Almost certainly don't need the `this` pointer.